@here/harp-mapview/lib/Utils.ts

```diff
@@ -330,6 +333,53 @@ export namespace MapViewUtils {
         }
     }
 
+    /**
+     * Return camera tilt angle (in degrees).
+     *
+     * Tilt is the angle between camera __look at__ (forward) vector and ground normal.
+     *
+     * @note It may depend on the projection type.
+     *
+     * @param camera The camera used for tilt angle calculation.
+     * @param projection The projection used to convert geo coordinated to world coordinates.
+     * @returns angle in radians.
+     */
+    export function getCameraTiltAngle(
+        camera: THREE.Camera,
+        projection: geoUtils.Projection
+    ): number {
+        // Note using different temporary vector then internal function used.
+        const lookAt: THREE.Vector3 = tmpVectors[0];
+        camera.getWorldDirection(lookAt);
+        return lookAt.angleTo(getGroundNormalInv(camera.position, projection));
+    }
+
+    /**
+     * Return normal to the ground surface directly above camera position.
+     *
+     * @note Function is not exposed cause it returns internal temporary vectors that
+     * can not be modified. Use with care and clone the result if required.
+     *
+     * @param position The position above the ground for which normal is calculated.
+     * @param projection The projection used to transform geo coordinates to world space.
+     */
+    function getGroundNormalInv(
+        position: THREE.Vector3,
+        projection: geoUtils.Projection
+    ): THREE.Vector3 {
+        // Position on the ground for surface normal calculation does not matter for
```

Comment:
This statement is not true. For example is the normal of the mercatorProjection (0,0,1) whereas the normal of the webMercatorProjection is (0,0,-1). Despite the fact that this is quite confusing I would rather always take the code path below.

Reply:
Ok, good to know!

Reply:
Will be fixed in follow up commit.